hris_publisher/hris.py

```diff
@@ -0,0 +1,187 @@
+
+    def _read_object(self, streaming_body):
+        """Return the content of the file object byte stream we retrieved."""
+
+        return json.loads(streaming_body.read())
+
+
+class Groups(object):
+    """Return a list of groups for a given profile based on rules."""
+    def __init__(self, entry):
+        self.hris_entry = entry
+        self.hris_grouplist = []
+
+    @property
+    def all(self):
+        """Run all rules and return the grouplist from the constructor."""
+        self.cost_center_rule()
+        self.cost_center_hierarchy()
+        self.management_level_rule()
+        self.manager_name_rule()
+        self.manager_status_rule()
+        self.egencia_country_rule()
+
+        return self.hris_grouplist
+
+    def cost_center_rule(self):
+        """Assert group based on splatting cost center."""
+        # Will be number only
+
+        full_cost_center = self.hris_entry.get('Cost_Center')
+        cost_center_code = full_cost_center.split(' ')[0]
+
+        group_name = 'hris_costcenter_{id}'.format(id=cost_center_code)
+        return self._add_group(group_name)
+
+    def cost_center_hierarchy(self):
+        """Dept groups (really)"""
+        dept = self._replace_spaces(self.hris_entry.get('Cost_Center_Hierarchy'))
+        group_name = 'hris_dept_{dept}'.format(dept=dept)
+        return self._add_group(group_name)
+
+    def management_level_rule(self):
+        """Assert group based on management level."""
+        management_level = self.hris_entry.get('Management_Level')
+
+        group_name = 'hris_{mangement_level}'.format(mangement_level=self._replace_spaces(management_level))
+        return self._add_group(group_name)
+
+    def manager_name_rule(self):
+        """Add to group for direct manager."""
+        manager_name = self.hris_entry.get('WorkersManager')
+        manager_name = self._replace_spaces(manager_name)
+
+        group_name = 'hris_direct_reports_{manager_name}'.format(manager_name=manager_name)
+
+        return self._add_group(group_name)
+
+    def manager_status_rule(self):
+        """Add to group is isManager"""
+        managerStatus = self.hris_entry.get('IsManager')
+        if managerStatus == 'TRUE':
+            group_name = 'hris_managers'
+        else:
+            group_name = 'hris_nonmanagers'
+        return self._add_group(group_name)
+
+    def egencia_country_rule(self):
+        country = self.hris_entry.get('EgenciaPOSCountry')
+        group_name = 'hris_egencia_{country}'.format(country=country)
+        return self._add_group(group_name)
+
+    def _add_group(self, group_name):
+        group_name = self._to_lower(group_name)
+
+        if group_name is not None:
+            self.hris_grouplist.append(group_name)
+            return group_name
+        else:
+            return None
+
+    def _replace_spaces(self, input_string):
+        if input_string:
+            return input_string.replace(" ", "_")
+
+    def _to_lower(self, input_string):
```

Comment:
complete nit but.. why not just: `'BLABLA'lower()` instead of `self._to_lower('BLABLA')`  each time?

Reply:
It was just an attempt at avoiding duplicate code.